services/src/datasets/external/natur40.rs

```diff
@@ -0,0 +1,290 @@
+                    ..DatasetOptions::default()
+                },
+            )?;
+            Ok(dataset)
+        })
+        .await
+        .context(error::TokioJoin)?
+    }
+
+    async fn load_raster_dbs(&self) -> Result<RasterDbs> {
+        Client::new()
+            .get(format!("{}/rasterdbs.json", self.base_url))
+            .basic_auth(&self.user, Some(&self.password))
+            .send()
+            .await?
+            .json()
+            .await
+            .context(error::Reqwest)
+    }
+}
+
+#[async_trait]
+impl MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for Natur40DataProvider
+{
+    async fn meta_data(
+        &self,
+        dataset: &DatasetId,
+    ) -> Result<
+        Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>>,
+        geoengine_operators::error::Error,
+    > {
+        let dataset = dataset
+            .external()
+            .ok_or(geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+            })?;
+        let split: Vec<_> = dataset.dataset_id.split(':').collect();
+
+        let (db_name, band_index) = match *split.as_slice() {
+            [db, band_index] => {
+                if let Ok(band_index) = band_index.parse::<isize>() {
+                    (db, band_index)
+                } else {
+                    return Err(geoengine_operators::error::Error::LoadingInfo {
+                        source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+                    });
+                }
+            }
+            _ => {
+                return Err(geoengine_operators::error::Error::LoadingInfo {
+                    source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+                })
+            }
+        };
+
+        let db_url = RasterDb::url_from_name(&self.base_url, db_name);
+        let dataset = self.load_dataset(db_url.clone()).await.map_err(|e| {
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })?;
+
+        Ok(Box::new(GdalMetaDataStatic {
+            time: None,
+            params: gdal_parameters_from_dataset(&dataset, band_index, Path::new(&db_url))?,
+            result_descriptor: raster_descriptor_from_dataset(&dataset, band_index, None)?,
+        }))
+    }
+}
+
+#[async_trait]
+impl
+    MetaDataProvider<MockDatasetDataSourceLoadingInfo, VectorResultDescriptor, VectorQueryRectangle>
+    for Natur40DataProvider
+{
+    async fn meta_data(
+        &self,
+        _dataset: &DatasetId,
+    ) -> Result<
+        Box<
+            dyn MetaData<
+                MockDatasetDataSourceLoadingInfo,
+                VectorResultDescriptor,
+                VectorQueryRectangle,
+            >,
+        >,
+        geoengine_operators::error::Error,
+    > {
+        todo!()
```

Comment:
This panics, maybe return an error instead?

```suggestion
        Err(geoengine_operators::error::Error::NotYetImplemented)
```

Reply:
f65088f41f89f69c260284e8f82ee3c3b851bf5f